### ISSUE TITLE
Error running rabbitmq during docker compose (permission denied)

### ISSUE THREAD

**Issue by crow-ua** [dotnet-architecture/eShopOnContainers#579]
When I'm trying to run the solution from the Visual Studio, docker compose fails to start up rabbitmq: 

`1>Creating network "dockercompose10799215418166897880_default" with the default driver
1>Creating dockercompose10799215418166897880_rabbitmq_1 ...
1>Creating dockercompose10799215418166897880_nosql.data_1 ...
1>Creating dockercompose10799215418166897880_basket.data_1 ...
1>Creating dockercompose10799215418166897880_webstatus_1   ...
1>Creating dockercompose10799215418166897880_sql.data_1    ...
1>Creating dockercompose10799215418166897880_webstatus_1   ... done
1>Creating dockercompose10799215418166897880_sql.data_1    ... done
1>Creating dockercompose10799215418166897880_identity.api_1 ...
1>Creating dockercompose10799215418166897880_nosql.data_1   ... done
1>Creating dockercompose10799215418166897880_rabbitmq_1     ... error
1>ERROR: for dockercompose10799215418166897880_rabbitmq_1  Cannot start service rabbitmq: driver failed programming external connectivity on endpoint dockercompose10799215418166897880_rabbitmq_1 (420e82842d5499e67520bbf9779e25d4db5d1dcdf98c8b8a1257bd0ed10b87d3): Error starting userland proxy: Bind for 0.0.0.0:5672: unexpected error Permission denied
1>Creating dockercompose10799215418166897880_basket.data_1  ... done
1>Creating dockercompose10799215418166897880_identity.api_1 ... done
1>ERROR: for rabbitmq  Cannot start service rabbitmq: driver failed programming external connectivity on endpoint dockercompose10799215418166897880_rabbitmq_1 (420e82842d5499e67520bbf9779e25d4db5d1dcdf98c8b8a1257bd0ed10b87d3): Error starting userland proxy: Bind for 0.0.0.0:5672: unexpected error Permission denied
1>Encountered errors while bringing up the project.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_SERVICE_BUS_USERNAME variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_SERVICE_BUS_PASSWORD variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The INSTRUMENTATION_KEY variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ORCHESTRATOR_TYPE variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_AZURE_STORAGE_CATALOG_NAME variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_AZURE_STORAGE_CATALOG_KEY variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_AZUREFUNC_CAMPAIGN_DETAILS_URI variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_AZURE_STORAGE_MARKETING_NAME variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : The ESHOP_AZURE_STORAGE_MARKETING_KEY variable is not set. Defaulting to a blank string.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating network "dockercompose10799215418166897880_default" with the default driver
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_rabbitmq_1 ... 
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_nosql.data_1 ... 
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_basket.data_1 ... 
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_webstatus_1   ... 
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_sql.data_1    ... 
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_webstatus_1   ... done
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_sql.data_1    ... done
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_identity.api_1 ... 
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_nosql.data_1   ... done
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_rabbitmq_1     ... error
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : ERROR: for dockercompose10799215418166897880_rabbitmq_1  Cannot start service rabbitmq: driver failed programming external connectivity on endpoint dockercompose10799215418166897880_rabbitmq_1 (420e82842d5499e67520bbf9779e25d4db5d1dcdf98c8b8a1257bd0ed10b87d3): Error starting userland proxy: Bind for 0.0.0.0:5672: unexpected error Permission denied
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_basket.data_1  ... done
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Creating dockercompose10799215418166897880_identity.api_1 ... done
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : ERROR: for rabbitmq  Cannot start service rabbitmq: driver failed programming external connectivity on endpoint dockercompose10799215418166897880_rabbitmq_1 (420e82842d5499e67520bbf9779e25d4db5d1dcdf98c8b8a1257bd0ed10b87d3): Error starting userland proxy: Bind for 0.0.0.0:5672: unexpected error Permission denied
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(363,5): error : Encountered errors while bringing up the project..`

I've googled that this error may indicate the port is already in use, but both `docker ps` and `netstat -a -n | grep 5672` does not indicate there are conflicts. 
Running `docker run rabbitmq:3-management-alpine` from PowerShell succeeds.
Restarting Visual Studio as administrator did not help. Changing rabbitmq to `3.7.4-management-alpine` also did not have any positive effect.
@CESARDELATORRE

**Comment by CESARDELATORRE** [comment]
This should not happen in a regular/clean dev machine that has been prepared as explained in the Wiki.
First of all it would be good if you can try it in a different machine so you make sure it is a local environment issue.
I'd try the following:
- Make sure you configured Docker as explained in the Wiki, providing enough memory to Docker (at least 4GB memory for Docker) since the SQL Server plus all the infrastructure containers need significant memory: 
https://github.com/dotnet-architecture/eShopOnContainers/wiki/02.-Setting-eShopOnContainers-in-a-Visual-Studio-2017-environment#installing-and-configuring-docker-in-your-development-machine
Sometimes the errors are not exactly the cause..
- If the error's cause is the network, make sure the Hyper-V Network Hub is working properly. Can you deploy any other Docker app without any issue?, just to isolate if the issue is with Docker or with just the RabbitMQ container.

Related to that, What is your environment and versions? - Windows 10? - Mac?

I think the best initial thing to do is to deploy it into a second "clean machine" and compare that new environment that should work with the environment where it is failing.

**Comment by mvelosop** [comment]
Hi @crow-ua, I had a situation like this once, and the cause was that I had run the app from the CLI and then opened VS, and that started the docker-compose automatically in VS.

The solution, @CESARDELATORRE found out, is just disabling auto pulling images and auto starting containers in VS:

![image](https://user-images.githubusercontent.com/7357806/38326610-1eaa6fb2-383e-11e8-837e-5387394dc592.png)

Just in case this is the situation!

I also have a problem with my docker networking and always have to restart docker after powering up my laptop.

**Comment by crow-ua** [comment]
@CESARDELATORRE 
Yeah, I did that (other machine) and everything worked like a charm from scratch. I guess I need to reset/reinstall Docker. The fact why container starts from CLI with the same settings is still a mystery.
FWIW, my main workstation is Win10 Enterprise, VS 2017, 32 Gb RAM.
@mvelosop 
I did not use CLI prior to VS, but thanks for information, that may be helpful in the future.
Closing the issue.
